### PR TITLE
Use svg <animate> instead of css

### PR DESCRIPTION
**Pull request by lixiang810**
#13 

> - [x] We can refactor some \<animate /\> into independent variables.

I don't know whether I did it correctly or not.

### CODE REVIEW

**Comment by ardeora** [comment]
Ooh this is very interesting! I am at work right now but I will take a look at this once I get off. Maybe we can use this to eliminate the goober dependency all together? The only thing to take care of is entry and exit animation on the container element. I was thinking we can use a ref and manually add those animation styles using the `animate` API https://developer.mozilla.org/en-US/docs/Web/API/Element/animate Support across browser seems good?

**Comment by lixiang810** [comment]
> I was thinking we can use a ref and manually add those animation styles using the `animate` API

Done in lixiang810/solid-toast@12475ce.

**Comment by ardeora** [comment]
Woah! This looks awesome! Sorry for the late reply. I will be reviewing the changesnow. I will get back to you in an hour or so. Thanks so much once again!

File: src/components/SuccessIcon.tsx [hunk 1]
```diff
@@ -1,50 +1,33 @@
...
-          'transform-origin': '50% 50%',
-          'animation-delay': '250ms',
-        }}
-        fill={props.primary || '#34C759'}
-        cx="16"
-        cy="16"
-        r="12"
-      />
+      <MainCircle fill={fill} />
+      <SecondaryCircle fill={fill} begin="250ms" />
```
Comment (ardeora) [inline src/components/SuccessIcon.tsx:10]:
```suggestion
      <SecondaryCircle fill={fill} begin="350ms" />
```
Small delay change

File: src/components/SuccessIcon.tsx [hunk 2]
```diff
@@ -1,50 +1,33 @@
...
+        stroke-dashoffset="22"
         stroke-linecap="round"
         stroke-miterlimit="10"
         d="M9.8,17.2l3.8,3.6c0.1,0.1,0.3,0.1,0.4,0l9.6-9.7"
-      />
+      >
+        <animate
+          attributeName="stroke-dashoffset"
+          values="22;0"
+          dur="0.2s"
```
Comment (ardeora) [inline src/components/SuccessIcon.tsx:24]:
```suggestion
          dur="0.25s"
```
Minor duration changes

File: src/components/ToastBar.tsx [hunk 3]
```diff
@@ -1,36 +1,46 @@
...
+    const direction = d(props.toast, props.position);
+    if (props.toast.visible) {
+      el.animate(
+        [
+          { transform: `translate3d(0,${direction * -200}%,0) scale(.6)`, opacity: 0.5 },
+          { transform: 'translate3d(0,0,0) scale(1)', opacity: 1 },
+        ],
+        {
+          duration: 250,
+          fill: 'forwards',
```
Comment (ardeora) [inline src/components/ToastBar.tsx:20]: Adding cubic bezier easing function
```suggestion
          duration: 350,
          fill: 'forwards',
          easing: 'cubic-bezier(.21,1.02,.73,1)'
```
Adding easing timing to animation

File: src/components/ToastBar.tsx [hunk 4]
```diff
@@ -1,36 +1,46 @@
...
+    } else {
+      el.animate(
+        [
+          { transform: 'translate3d(0,0,-1px) scale(1)', opacity: 1 },
+          { transform: `translate3d(0,${direction * -150}%,-1px) scale(.4)`, opacity: 0 },
+        ],
+        {
+          duration: 400,
+          fill: 'forwards',
+        }
```
Comment (ardeora) [inline src/components/ToastBar.tsx:32]:
```suggestion
          fill: 'forwards',
          easing: 'cubic-bezier(.06,.71,.55,1)'
        }
```

Same easing addition

**Review by ardeora — COMMENTED** [review]
Hey there! thanks again for taking out time to do this. All these changes are awesome. Removing Goober will shave off 1kb from the gzipped package size 🥳🥳 I have some small animation and style adjustments that will help make the animations smoother

**Comment by ardeora** [comment]
This looks great! Will push a release soon! Thanks for all the amazing work on this

**Comment by lixiang810** [comment]
Goober was removed at lixiang810@5e387c1, I think you would like to review that.

**Comment by ardeora** [comment]
I just checked the actual production size of solid-toast and it went from 5kb to 3.82kb 🤯

https://bundlejs.com/?q=skypack%3Asolid-toast&treeshake=%5B%7B+Toaster%2Ctoast+%7D%5D&config=%7B%22analysis%22%3Atrue%2C%22esbuild%22%3A%7B%22external%22%3A%5B%22solid-js%22%5D%7D%7D

**Pull request merged**